components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/JWTTokenIssuer.java

```diff
@@ -622,10 +622,11 @@ protected JWTClaimsSet createJWTClaimSet(OAuthAuthzReqMessageContext authAuthzRe
         JWTClaimsSet jwtClaimsSet;
 
         // Handle custom claims
+        // TODO : We need to handle jwt access token claims issuing for old and new application.
         if (authAuthzReqMessageContext != null) {
             jwtClaimsSet = handleCustomClaims(jwtClaimsSetBuilder, authAuthzReqMessageContext);
         } else {
-            jwtClaimsSet = handleCustomClaims(jwtClaimsSetBuilder, tokenReqMessageContext);
+            jwtClaimsSet = handleJWTAccessTokenOIDCClaims(jwtClaimsSetBuilder, tokenReqMessageContext);
         }
```

Comment:
AT related claim set needs to be handled for both when AT is issued over authorization flow (e.g: implicit) and token flow.
As a delivery point of view we can take a decision we handle this for token flow only. But this needs to be improved

Reply:
Yes, will handle both here. current implementation only focuses on the token flow.